### PR TITLE
refactor: use config-loader

### PR DESCRIPTION
**Pull request by shellscape**
\+ update to latest eslint-config-webpack

This PR contains a:

- [ ] **bugfix**
- [ ] new **feature**
- [x] **code refactor**
- [x] **test update** 
- [ ] **typo fix**
- [x] **metadata update**

### Motivation / Use-Case

`@webpack-contrib/config-loader` provides many advantages for loading webpack configs, including supporting all export formats of config that can be used out of the gate by `webpack-cli` and `webpack-command`. It also allows for require hooks for loading configs, and for different file formats for config files. 

### Breaking Changes

None

### CODE REVIEW

File: cli.js [hunk 1]
```diff
@@ -14,7 +12,7 @@ const cosmiconfig = require('cosmiconfig');
 const debug = require('debug')('webpack-serve');
 const meow = require('meow');
 const merge = require('lodash/merge');
-const resolveModule = require('resolve').sync;
+// const resolveModule = require('resolve').sync;
```
Comment (ooflorent) [inline cli.js]: Could you remove the comment?

File: cli.js [hunk 2]
```diff
@@ -59,61 +58,33 @@ const cli = meow(chalk`
...
+        type: 'string',
+      },
+    },
   }
-});
+);
 
 const flags = Object.assign({}, cli.flags);
 
-if (flags.require) {
```
Comment (ooflorent) [inline cli.js]: Why did you remove this option?

File: lib/options.js [hunk 3]
```diff
@@ -1,10 +1,8 @@
-'use strict';
-
-const path = require('path');
+// const path = require('path');
```
Comment (ooflorent) [inline lib/options.js]: Not needed

File: cli.js (hunk 2 above)
Comment (shellscape) [reply cli.js]: it's handled by `config-loader` downstream

**Comment by shellscape** [comment]
@evilebottnawi it does not break `serve` options being in `webpack.config.js`, no.
